### ISSUE TITLE
no-var vs prefer-const word/token/column reporting is inconsistent

### ISSUE THREAD

**Issue by sparr** [eslint/eslint#9402]
**What rule do you want to change?** `prefer-const`

**Does this change cause the rule to produce more or fewer warnings?** same or fewer

**How will the change be implemented? (New option, new default behavior, etc.)?** new default behavior

**Please provide some example code that this change will affect:**

```js
var a = 2;
let b = 3, e = 4;
const c = 4;
const d = a+b+c+e;
```

**What does the rule currently do for this code?**

`b` (col 5) and `e` (col 12) are identified in the second line as never reassigned

**What will the rule do after it's changed?**

`let` (col 1) would be identified when every declaration in the statement should be `const`

**Comment by platinumazure** [comment]
I'm not sure we can realistically make a change here. You can declare multiple variables in a `let` statement, and if a subset of them are reassigned, they should not be flagged by `prefer-const`:

```js
let a = 5,  // a should be const
    b = 6,  // b should NOT be const
    c = 10; // c should be const

b = 1;
```

Whereas with `no-var`, all variable declarations of kind `var` can be switched to `let`, so we report it at the statement level.

**Comment by sparr** [comment]
Could it target the `let` when there's just one variable, or when every variable in the statement should be `const`?

**Comment by platinumazure** [comment]
Personally that just seems like it would complicate the reporting logic of the rule for not much benefit (aesthetic only, situational). I don't think it's worthwhile. But I'm open to being persuaded by the rest of the team.

**Comment by platinumazure** [comment]
@sparr Would you mind editing your original post to reflect the possible enhancement request (`prefer-const` targeting the entire statement/`let` keyword when all variables in the statement could/should be `const`)?

If you could use the [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md) as you do so, that would be really appreciated. Thanks!

**Comment by sparr** [comment]
@platinumazure done. ?

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
